plugins/source/aws/resources/services/fsx/data_repo_tasks.go

```diff
@@ -0,0 +1,170 @@
+			},
+			{
+				Name:        "report_format",
+				Description: "Required if Enabled is set to true",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Report.Format"),
+			},
+			{
+				Name:        "report_path",
+				Description: "Required if Enabled is set to true",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Report.Path"),
+			},
+			{
+				Name:        "report_scope",
+				Description: "Required if Enabled is set to true",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("Report.Scope"),
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) for a given resource",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ResourceARN"),
+			},
+			{
+				Name:        "start_time",
+				Description: "The time that Amazon FSx began processing the task",
+				Type:        schema.TypeTimestamp,
+			},
+			{
+				Name:        "status_failed_count",
+				Description: "A running total of the number of files that the task failed to process",
+				Type:        schema.TypeBigInt,
+				Resolver:    schema.PathResolver("Status.FailedCount"),
+			},
+			{
+				Name:        "status_last_updated_time",
+				Description: "The time at which the task status was last updated",
+				Type:        schema.TypeTimestamp,
+				Resolver:    schema.PathResolver("Status.LastUpdatedTime"),
+			},
+			{
+				Name:        "status_succeeded_count",
+				Description: "A running total of the number of files that the task has successfully processed",
+				Type:        schema.TypeBigInt,
+				Resolver:    schema.PathResolver("Status.SucceededCount"),
+			},
+			{
+				Name:        "status_total_count",
+				Description: "The total number of files that the task will process",
+				Type:        schema.TypeBigInt,
+				Resolver:    schema.PathResolver("Status.TotalCount"),
+			},
+			{
+				Name:        "tags",
+				Description: "A list of Tag values, with a maximum of 50 elements",
+				Type:        schema.TypeJSON,
+				Resolver:    resolveDataRepoTasksTags,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchFsxDataRepoTasks(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().FSX
+	input := fsx.DescribeDataRepositoryTasksInput{MaxResults: aws.Int32(1000)}
+	for {
+		result, err := svc.DescribeDataRepositoryTasks(ctx, &input)
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- result.DataRepositoryTasks
+		if aws.ToString(result.NextToken) == "" {
+			break
+		}
+		input.NextToken = result.NextToken
+	}
+	return nil
+}
```

Comment:
I think you can use `fsx.NewDescribeDataRepositoryTasksPaginator` here, which should make the code a little simpler since we then don't need to handle the `NextToken` here

Reply:
Great suggestion, thanks!